sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -231,21 +233,37 @@ def __next__(self):
         if self._complete:
             raise StopIteration("Download complete")
         if not self._iter_downloader:
-            # If no iterator was supplied, the download completed with
-            # the initial GET, so we just return that data
+            # cut the data obtained from initial GET into chunks
+            if len(self._current_content) > self._chunk_size:
+                return self._get_chunk_data()
             self._complete = True
             return self._current_content
 
         if not self._iter_chunks:
             self._iter_chunks = self._iter_downloader.get_chunk_offsets()
-        else:
+
+        # initial GET result still has more than _chunk_size bytes of data
+        if len(self._current_content) >= self._chunk_size:
+            return self._get_chunk_data()
+
+        try:
             chunk = next(self._iter_chunks)
-            self._current_content = self._iter_downloader.yield_chunk(chunk)
+            self._current_content += self._iter_downloader.yield_chunk(chunk)
+        except StopIteration as e:
+            self._complete = True
+            if self._current_content:
+                return self._current_content
+            raise e
 
-        return self._current_content
+        return self._get_chunk_data()
```

Comment:
What's the reason for calling `self._get_chunk_data()` in this case? as far as I understand if we hit this case (we're done iterating the first single get call and now iterating through our download chunks like we usually do) then current content should always be the same as the chunk size so we can just return `self.current_content`. It seems from your `try-except` that there is an edge case where there will be excess content - could you just put a small comment on that in the code - its not obvious to me haha

EDIT: I just realized the case is when the current content from the first get is still there but smaller than chunk size therefore we want to make sure its also included. I think adding a comment in the code would be very helpful here!
Good work thinking of all the cases